src/modules/menu/components/share/MenuShareInputCopyButton.vue

```diff
@@ -1,17 +1,119 @@
+<script setup>
+import { computed, onBeforeUnmount, onMounted, onUpdated, ref, watch } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+// import { useStore } from 'vuex'
+import { useTippyTooltip } from '@/utils/composables/useTippyTooltip'
+
+const props = defineProps({
+    inputText: {
+        type: String,
+        default: null,
+    },
+    small: {
+        type: Boolean,
+        default: true,
+    },
+    copyText: {
+        type: String,
+        default: 'copy_cta',
+    },
+    copiedText: {
+        type: String,
+        default: 'copy_done',
+    },
+    labelText: {
+        type: String,
+        default: null,
+    },
+    hasWarning: {
+        type: Boolean,
+        default: false,
+    },
+})
+
+const copiedInClipboard = ref(false)
+const timeoutCopied = ref(null)
+
+// const store = useStore()
+const i18n = useI18n()
+// const store = useStore()
+
+const { refreshTippyAttachment, removeTippy } = useTippyTooltip(
+    '#input-copy-button[data-tippy-content]',
+    {
+        placement: 'top',
+        offset: [0, -20],
+    }
+)
+
+const buttonText = computed(() => {
+    return i18n
+        .t(copiedInClipboard.value ? props.copiedText : props.copyText)
+        .replace('&nbsp;', '\xa0')
+})
+// const isPhoneMode = computed(() => store.getters.isPhoneMode)
+
+const clearIsCopiedInClipboard = () => {
+    copiedInClipboard.value = false
+    clearTimeout(timeoutCopied.value)
+}
+
+const copyInputToClipboard = () => {
+    navigator.clipboard.writeText(props.inputText)
+    copiedInClipboard.value = true
+    timeoutCopied.value = setTimeout(clearIsCopiedInClipboard, 2500)
+}
+
+watch(
+    () => props.inputText,
+    () => {
+        clearIsCopiedInClipboard()
+    }
+)
+
+onMounted(() => {
+    refreshTippyAttachment()
+})
+
+onBeforeUnmount(() => {
+    clearTimeout(timeoutCopied.value)
+})
+
+onUpdated(() => {
+    refreshTippyAttachment()
+    if (!props.hasWarning) {
+        removeTippy()
+    } else {
+        refreshTippyAttachment()
+    }
+})
+</script>
+
 <template>
-    <div v-if="inputText">
+    <div
+        v-if="inputText"
+        id="input-copy-button"
+        data-tippy-content="warn_share_local_file"
```

Comment:
the tooltip color should match the warning from the button (add a new tooltip style if necessary)